esiclient/v1/orchestrator/cluster.py

```diff
@@ -222,3 +223,80 @@ def take_action(self, parsed_args):
 
         return ["Node", "Port", "Network", "Fixed IP",
                 "Floating Network", "Floating IP"], data
+
+
+class Undeploy(command.Command):
+    """Undeploy a cluster from ESI nodes"""
+
+    log = logging.getLogger(__name__ + ".Undeploy")
+    REQUIRED_FIELDS = ['nodes', 'private_network_name', 'api_vip',
+                       'ingress_vip']
+
+    def get_parser(self, prog_name):
+        parser = super(Undeploy, self).get_parser(prog_name)
+
+        parser.add_argument(
+            "cluster_config_file",
+            metavar="<cluster_config_file>",
+            help=_("File describing the cluster configuration"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        cluster_config_file = parsed_args.cluster_config_file
+        with open(cluster_config_file) as f:
+            cluster_config = json.load(f)
+
+        print("STARTING UNDEPLOY")
+
+        ironic_client = self.app.client_manager.baremetal
+        neutron_client = self.app.client_manager.network
+        node_configs = cluster_config['node_configs']
+
+        uuid_node_configs = [node_config for node_config in node_configs
+                             if 'node_uuids' in node_config['nodes']]
+        count = 0
+        for node_config in uuid_node_configs:
+            count += 1
+            print("* undeploying %s of %s node configs" % (
+                count, len(uuid_node_configs)))
+            node_uuids = node_config['nodes']['node_uuids']
+            network_config = node_config['network']
+            network_uuid = network_config.get('network_uuid', None)
+            network = neutron_client.find_network(network_uuid)
+            for node in node_uuids:
+                print("   * %s" % node)
+                ironic_client.node.set_provision_state(node, 'deleted')
+                print("   * waiting for nodes to start undeploy before"
```

Comment:
Will you get an error if deleting a port while a node is still under deleting?

Reply:
Nope - but it takes ~10 seconds for Ironic to detach the port itself, so I figure there's no harm in waiting!